pkg/certificate/service.go

```diff
@@ -71,10 +74,58 @@ func New(ctx context.Context, idCfg *config.IdentityConfig) (daemon.Daemon, erro
 	var localFileKeyPEM []byte
 	var localFileIdentity *InstanceIdentity
 
+	// validate file
+	isValidFile := func(file_path string) error {
+		dir_path := filepath.Dir(file_path)
+		var target_path string
+		_, file_err := os.Stat(file_path)
+		_, dir_err := os.Stat(dir_path)
+
+		if file_err == nil {
+			// validate file path
+			target_path = file_path
+		} else if os.IsNotExist(file_err) && dir_err == nil {
+			// validate dir path
+			target_path = dir_path
+		} else {
+			return fmt.Errorf("file path not exist: %w, %w", file_err, dir_err)
+		}
+
+		err := unix.Access(target_path, unix.W_OK)
+		if err != nil {
+			return fmt.Errorf("file permission error: %w", err)
+		}
+
+		return nil
+	}
+
+	// validate files
+	isValidFiles := func() error {
+		if !idCfg.ServiceCert.LocalCert.Use {
+			for _, certFile := range idCfg.ServiceCert.CopperArgos.Cert.Paths {
+				err := isValidFile(certFile)
+				if err != nil {
+					return err
+				}
+			}
+			for _, keyFile := range idCfg.ServiceCert.CopperArgos.Key.Paths {
+				err := isValidFile(keyFile)
+				if err != nil {
+					return err
+				}
+			}
+			err := isValidFile(idCfg.CaCertFile)
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	}
```

Comment:
Do the following:
- Do early return as much as possible => gives less indent = more readable code = less buggy as easier to fix
- In go, you can return error without defining them in each line.

```suggestion
  isValidFiles := func() error {
		// TODO: Write a reason why LocalCert Mode does not require the cert path validation!!
		if idCfg.ServiceCert.LocalCert.Use {
			return nil
		}

		for _, certFile := range idCfg.ServiceCert.CopperArgos.Cert.Paths {
			if err := isValidFile(certFile); err != nil {
				return err
			}
		}

		for _, keyFile := range idCfg.ServiceCert.CopperArgos.Key.Paths {
			if err := isValidFile(keyFile); err != nil {
				return err
			}
		}

		return isValidFile(idCfg.CaCertFile) // will return error if idCfg.CaCertFile is not a valid filepath
	}
```